cowprotocol/cowswap
# feat(smart-slippage): improve ux

# Summary

## Update 2024-09-03

- Add `suggested` when smart slippage differs from user selection

![Screenshot 2024-09-03 at 17 58 46](https://github.com/user-attachments/assets/3d111173-18c9-49a9-95ce-234a50690d98)

- On click, slippage value is updated to `auto`, using the smart slippage
- Add `dynamic` to slippage when it's in use (both on form and confirm modal)

![Screenshot 2024-09-03 at 17 59 07](https://github.com/user-attachments/assets/34c32dc8-6625-4663-b4f5-57db44e4abee)
![Screenshot 2024-09-03 at 17 59 38](https://github.com/user-attachments/assets/331d6047-1525-46e0-a15a-1b37d0c35246)

- When user input value is same as smart value, do not show warnings

![Screenshot 2024-09-03 at 17 59 20](https://github.com/user-attachments/assets/b0f248ba-96f7-477c-aaa4-8bf4221a314f)

⚠️ Texts, behaviour are still up to discussion.
⚠️ EthFlow remains as is as per Felix L opinion, to prevent too many non-refundable orders.

# Testing

1. Make sure the feature flag is on
2. Set slippage to `auto` if not already
3. Select a pair, fill in trade details
4. Expand the fee dropdown
* Slippage should be `dynamic`
5. Click on it to open the settings modal
* Should show the `dynamic` text and tooltip
6. Update it to match the exact value suggested
* `dynamic` should disappear from the form and the settings
* There should be no warnings
7. Change to a value other than the suggested percentage
* The form should show `suggested` link next to the current slippage
8. Click on it
* Should set slippage to `auto` and form should show `dynamic` next to slippage
9. On confirmation modal:
* `dynamic` should be shown when slippage set to `auto`
* not dynamic when manually set to anything else



## Original
Follow up on https://github.com/cowprotocol/cowswap/pull/4819

Quick draft to see if we can always show the default slippage and allow to reset the modified slippage with the default

<img width="1446" alt="image" src="https://github.com/user-attachments/assets/38a98c84-8031-4411-b5f7-0b580c99427a">


WIP: Unsure if I will get this done before my leave
